scout/adapter/mongo/event.py

```diff
@@ -79,26 +79,45 @@ def events(self, institute, case=None, variant_id=None, level=None,
               list: filtered query returning matching events
         """
 
-        query = {'institute': institute['_id']}
+        query = {}
 
         if variant_id:
-            query['category'] = 'variant'
-            query['variant_id'] = variant_id
-            if level:
-                query['level'] = level
-                if level != 'global':
-                    query['case'] = case['_id']
-        elif panel:
-            query['panel'] = panel
-        # If no variant_id or panel we know that it is a case level comment
+            if comments:
+                # If it's comment-related event collect global and variant-specific comment events
+                query = {
+                    '$or': [
+                        {
+                            'category' : 'variant',
+                            'variant_id' : variant_id,
+                            'verb' : 'comment',
+                            'level' : 'global'
```

Comment:
Hi! I don't understand, you never use the `level` parameter so this function will always return all comments for a variant...
I think you should add some tests for this to be sure that the behaviour is as expected. This one is a bit tricky so ot would be nice to have tests for the future.

Reply:
Why don't use 'level' ? If it's specified in the query isn't it used? 
The way the query is written should make it return:
 * global comments for variants with that variant_id (regardless of case and institute) 
 * specific comments for that variant_id, from that case and that institute.

What it DOES NOT RETURN are specific comments for that variant id but for other cases.

I'll write some tests as soon as I have some time, I'm trying to fix the aligners bug today.

Reply:
Aha great! I misunderstood, you are right.